mesonbuild/backend/backends.py

```diff
@@ -236,27 +236,31 @@ def serialise_tests(self):
 
     def determine_linker(self, target, src):
         if isinstance(target, build.StaticLibrary):
-            if self.build.static_cross_linker is not None:
+            if target.is_cross:
                 return self.build.static_cross_linker
             else:
                 return self.build.static_linker
-        if len(self.build.compilers) == 1:
-            return self.build.compilers[0]
+        if target.is_cross:
+            compilers = self.build.cross_compilers
+        else:
+            compilers = self.build.compilers
+        if len(compilers) == 1:
+            return compilers[0]
```

Comment:
Hmmm. There was a reason I did that but the commit history does not say what it was. This should be detected at configure time, though, since it goes through all the sources of the target and at least one of them should be compilable.